vnext/Microsoft.ReactNative/Views/TouchEventHandler.cpp

```diff
@@ -449,49 +469,76 @@ bool TouchEventHandler::TagFromOriginalSource(
   return true;
 }
 
-winrt::IPropertyValue TouchEventHandler::TestHit(
-    const winrt::Collections::IVectorView<xaml::Documents::Inline> &inlines,
-    const winrt::Point &pointerPos,
-    bool &isHit) {
-  winrt::IPropertyValue tag(nullptr);
-
-  for (const auto &el : inlines) {
-    if (const auto span = el.try_as<xaml::Documents::Span>()) {
-      auto resTag = TestHit(span.Inlines().GetView(), pointerPos, isHit);
+bool IsCharacterBefore(winrt::TextPointer textPointer, winrt::Point point) {
+  const auto rect = textPointer.GetCharacterRect(winrt::LogicalDirection::Forward);
+  const auto bottom = rect.Y + rect.Height;
+
+  // The character Rect always has Width = 0, so we need to use the X-dimension
+  // of the next character on the same line. If the next character is not on
+  // the same line, we use the rightmost boundary of the TextBlock.
+  //
+  // The side-effect is that wrapped text may have additional valid hit box
+  // space at the end of a line, but it's better than the alternative of the
+  // hit box excluding the last character on a line.
+  const auto nextPointer = textPointer.GetPositionAtOffset(1, winrt::LogicalDirection::Forward);
+  auto right = textPointer.VisualParent().Width();
+  if (nextPointer != nullptr) {
+    const auto nextRect = nextPointer.GetCharacterRect(winrt::LogicalDirection::Forward);
+    if (rect.Y == nextRect.Y) {
+      right = nextRect.X;
+    }
+  }
 
-      if (resTag)
-        return resTag;
+  // The character is before the point if the Y-coordinate of the point is
+  // below (greater than) the bottom of the character rect, or if the
+  // Y-coordinate is below (greater than) the top of the character rect and
+  // the X-coordinate is greater than the right side of the character rect:
+  // ┌───────────┐
+  // │     ┌─────┘
+  // └─────┘ (x,y)
```

Comment:
funny because I had to implement this exact kind of logic for the xaml handwriting panel a few years ago :D 
This feels like something that should be part of the platform, but oh well, maybe one day :)